src/main/java/org/wildfly/security/auth/realm/token/TokenSecurityRealm.java

```diff
@@ -124,7 +124,17 @@ public boolean verifyEvidence(Evidence evidence) throws RealmUnavailableExceptio
                 return false;
             }
 
-            return strategy.validate((BearerTokenEvidence) evidence) != null;
+            BearerTokenEvidence tokenEvidence = BearerTokenEvidence.class.cast(evidence);
+
+            try {
+                return strategy.validate(tokenEvidence) != null;
+            } catch (RealmUnavailableException rue) {
+                throw rue;
+            } catch (Exception unknown) {
```

Comment:
We only expect ```RealmUnavailableException``` from token validators. Any other expcetion is unexpected.

I think your second suggestion makes more sense as we can have a single method used by both ```getClaims``` and ```verifyEvidence``` method.